### PR TITLE
MM-42267: Update browse channel UI

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
UI update for browse channel model. 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21173
JIRA: https://mattermost.atlassian.net/browse/MM-42267

#### Screenshots

|  before  |  after  |
|----|----|
| Screen Shot 2023-05-29 at 18 01 29 | Screen Shot 2023-05-29 at 16 43 00 |

#### Release Note
```release-note
Updated UI for browse channel modal
```

### CODE REVIEW

**Comment by sinansonmez** [comment]
As agreed with @hmhealey I have removed the member count from the modal as the current API to get info for each channel is quite heavy task. 

@matthewbirtch One question. As we have removed the member count, do we still need to show a small dot before channel description in the channels that we are not member (as an example, ut or veritatis channels in the before screenshot)?

**Comment by esethna** [comment]
Thanks @sinansonmez! Agree with removing the dot if there's nothing infront of it

**Review by esethna — APPROVED** [review]

**Comment by matthewbirtch** [comment]
> As agreed with @hmhealey I have removed the member count from the modal as the current API to get info for each channel is quite heavy task.
> 
> @matthewbirtch One question. As we have removed the member count, do we still need to show a small dot before channel description in the channels that we are not member (as an example, ut or veritatis channels in the before screenshot)?

Yep, makes sense to remove the dot If we're no longer showing the member count. I wasn't aware that we needed to do that, but if there are performance implications then so be it.

**Comment by sinansonmez** [comment]
> > As agreed with @hmhealey I have removed the member count from the modal as the current API to get info for each channel is quite heavy task.
> > @matthewbirtch One question. As we have removed the member count, do we still need to show a small dot before channel description in the channels that we are not member (as an example, ut or veritatis channels in the before screenshot)?
> 
> Yep, makes sense to remove the dot in this case if we're no longer showing the member count. I wasn't aware that we needed to do that, but if there are performance implications then so be it.

@matthewbirtch @esethna dot is removed if there is nothing before it

image

**Comment by sinansonmez** [comment]
@esethna can you request review from developers? As there were some internal changes in MM, I dont who should review this?

**Comment by sinansonmez** [comment]
@hmhealey any idea?

**Comment by hmhealey** [comment]
The build hash in the About Mattermost modal is `3b0d0d7179590e467e3225daddcb876e30ea8bf2` which is from master branch earlier today. I'll ask around to see if anyone knows why that might have happened. It's possible that it's because we finally changed the repo's name

**Comment by hmhealey** [comment]
@sinansonmez If you wouldn't mind resolving those conflicts and updating the branch to the latest master, that should be resolved. It was probably caused by the repo rename, but that should be fixed now

**Comment by sinansonmez** [comment]
@hmhealey I have updated it to latest master but test server is still not picking up the changes from the PR

**Comment by hmhealey** [comment]
That's very strange. I tried submitting my own branch with the same changes in case the problem was related to you being on a fork, but its test server is on the same commit. I've tried pushing an empty to see if that changes anything, and I'll also reach out to see if the team who had fixed that issue earlier in the week knows why it's still stuck on master

File: e2e-tests/cypress/tests/integration/channels/enterprise/accessibility/accessibility_modals_dialogs_spec.js
Comment (saturninoabril) [inline e2e-tests/cypress/tests/integration/channels/enterprise/accessibility/accessibility_modals_dialogs_spec.js:121]:
```suggestion
                    cy.get('#createNewChannelButton').focus().tab().tab().tab();
```

**Comment by sinansonmez** [comment]
thanks for running e2e tests @saturninoabril out of 3 failing tests I think 2 of them are related to my changes. I fixed them. Could you please run the tests again?
/update-branch

**Comment by sinansonmez** [comment]
@hmhealey test server is working correctly. @matthewbirtch please take a look

**Review by matthewbirtch — APPROVED** [review]
Lookin' good @sinansonmez! Thank you!

File: webapp/channels/src/components/more_channels/index.ts
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/more_channels/index.ts:1]: 0/5 but shouldn't this essentially called `browser_channels_modal` cc @hmhealey I know that would create a big diff, but being clear in naming is always a good fallback

File: webapp/channels/src/components/common/svg_images_components/magnifying_glass_svg.tsx
Comment (sinansonmez) [reply webapp/channels/src/components/common/svg_images_components/magnifying_glass_svg.tsx:1]: @M-ZubairAhmed it is this image. Therefore I dont think it is possible to import this from compass-icons.
Screenshot 2023-06-13 at 22 27 40

File: webapp/channels/src/components/more_channels/index.ts
Comment (sinansonmez) [reply webapp/channels/src/components/more_channels/index.ts:1]: 0/5 to me as well

Comment (hmhealey) [reply webapp/channels/src/components/more_channels/index.ts:1]: Let's leave it as-is for now since this PR's pretty big. We can do a followup once this is in though to rename it to `BrowseChannelsModal` after though since that's a good point that it's not super clear right now.

Historically, the modal was titled More Channels, but that's obviously changed since then. It's also less clear than some of our other historical names like the System Console being `AdminConsole` which is less ambiguous than `MoreChannels`

Comment (hmhealey) [reply webapp/channels/src/components/more_channels/index.ts:1]: Oh, the "More Channels" name was just the current name of the modal. I didn't realize it stayed that way even when we renamed the menu item to "Browse Channels". I guess I had just assumed they matched, and now I feel dumb for explaining the current state of it like it was some ancient MM fact 😛

File: webapp/channels/src/components/more_channels/more_channels.tsx
Comment (hmhealey) [inline webapp/channels/src/components/more_channels/more_channels.tsx:271]: I'm glad to see that we're getting rid of the original usage of `more-modal`. That's one of those overused classes in MM that makes working with our CSS styling difficult, so this'll definitely help to get rid of that eventually

**Review by hmhealey — APPROVED** [review]

File: webapp/channels/src/components/more_channels/index.ts
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/more_channels/index.ts:1]: Awesome make sense @sinansonmez would you mind opening a follow up PR after we merge this on on the name change ?

File: webapp/channels/src/components/common/svg_images_components/magnifying_glass_svg.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/common/svg_images_components/magnifying_glass_svg.tsx:1]: Ah yes I see yes you are right

File: webapp/channels/src/components/searchable_channel_list.tsx
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/searchable_channel_list.tsx:478]: I really hoped since this was a new component we could have written this in functional way but it's fine either ways. Also do you plan to remove this ?

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
🎉

File: webapp/channels/src/components/more_channels/index.ts
Comment (sinansonmez) [reply webapp/channels/src/components/more_channels/index.ts:1]: Yeah sure

**Comment by sinansonmez** [comment]
@M-ZubairAhmed i think this is ready to merge

**Pull request merged**
